### PR TITLE
Add watergate documentation

### PR DESCRIPTION
**Pull request by adam-the-hero**
## Proposed change

Introduce documentation for new Watergate integration.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [x] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [x] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository:  https://github.com/home-assistant/brands/pull/6054
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced comprehensive documentation for integrating Watergate Sonic devices with Home Assistant, detailing capabilities such as valve control, live telemetry monitoring, and water usage tracking.
- **Documentation**
	- Provided prerequisites and setup instructions for successful integration, including enabling the Local API and listing supported devices.

### CODE REVIEW

File: source/_integrations/watergate.markdown [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: Watergate
+description: Instructions on how to integrate Watergate within Home Assistant.
+ha_category:
+  - Sensor
+  - Valve
+  - Binary Sensor
+  - Number
+  - Water management
+ha_release: '2024.11'
```
Comment (joostlek) [inline source/_integrations/watergate.markdown]:
```suggestion
ha_release: '2025.1'
```

File: source/_integrations/watergate.markdown [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ha_iot_class: Local Push
+ha_config_flow: true
+ha_codeowners:
+  - '@watergate-ai'
```
Comment (joostlek) [inline source/_integrations/watergate.markdown]:
```suggestion
  - '@adam-the-hero'
```

File: source/_integrations/watergate.markdown [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ha_domain: watergate
+ha_platforms:
+  - diagnostics
+  - sensor
+  - valve
+  - binary_sensor
+  - number
```
Comment (joostlek) [inline source/_integrations/watergate.markdown]:
```suggestion
  - valve
```

**Review by joostlek — CHANGES_REQUESTED** [review]
To the docs reviewer, take the quality_scale.yml in account :) https://github.com/home-assistant/core/pull/129686/files#diff-3abed69396f1886d1f1b4f28e194aeb8bde4075780e32f91d3fe7c535221d277

**Comment by joostlek** [comment]
@adam-the-hero I am missing a link to the brands PR, please add it

File: source/_integrations/watergate.markdown [hunk 4]
```diff
@@ -0,0 +1,46 @@
+---
+title: Watergate
+description: Instructions on how to integrate Watergate within Home Assistant.
+ha_category:
+  - Sensor
+  - Valve
+  - Binary Sensor
+  - Number
+  - Water management
```
Comment (joostlek) [inline source/_integrations/watergate.markdown]: You only have Valve here as well

File: source/_integrations/watergate.markdown [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  - valve
+---
+
+The **Watergate** {% term integration %} integrates your Watergate Devices (currently Sonic) to your Home Assistant.
+With this integration you are able to:
+
+- control your valve
+- configure auto shut off parameters
+- gather auto shut off parameters
+- check diagnostics of the device
```
Comment (joostlek) [inline source/_integrations/watergate.markdown]: Please make sure the PR matches the current features implemented. As in, we should be able to cut a release at this moment and the documentation should be correct and not describing future features

**Review by joostlek — DISMISSED** [review]

**Comment by adam-the-hero** [comment]
To reviewers: I have created another PR adding more sensors hopefully we can do this together. https://github.com/home-assistant/core/pull/133015

File: source/_integrations/watergate.markdown [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  - sensor
+---
+
+The **Watergate** {% term integration %} integrates your Watergate Devices (currently Sonic) to your Home Assistant.
+With this integration you are able to:
+
+- control your valve
+- monitor telemetry
+- gather auto shut off reports
+- check diagnostics of the device
```
Comment (frenck) [inline source/_integrations/watergate.markdown]:
```suggestion
- Control your valve
- Monitor telemetry
- Gather auto shut-off reports
- Check diagnostics of the device
```

File: source/_integrations/watergate.markdown [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+## Prerequisites
+
+- You need to have Sonic device
+- Local API feature must be enabled in application
+- Network Traffic between Home Assistant Server and Sonic must be unlocked (Server uses HTTP protocol on 80 port).
```
Comment (frenck) [inline source/_integrations/watergate.markdown]:
```suggestion
- You need to have a Sonic device.
- Local API feature must be enabled in application.
- Network traffic between Home Assistant and Sonic must be possible. (server uses HTTP protocol on 80 port).
```

PS: I'm really doubting the latter point is an pre-requisite we should mention here.

File: source/_integrations/watergate.markdown [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+{% important %}
+Go to Watergate application and enable Local API in specific device's settings.
```
Comment (frenck) [inline source/_integrations/watergate.markdown]: This is a duplicate of the second point?

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/watergate.markdown (hunk 8 above)
Comment (adam-the-hero) [reply source/_integrations/watergate.markdown]: Yes it is. I've removed it I think there is no point to write 101 instructions for that.

**Comment by adam-the-hero** [comment]
Sorry for a lot of changes but I had to make this documentation more comprehensive. Now it has more information.

File: source/_integrations/watergate.markdown [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+#### Valves
+
+- **Water valve state**
+  - **Description**: The current state of the water valve (open/closed).
+  - **Remarks**: It is automatically updated when the valve state is changed.
+
+## Data updates
+
+The Watergate integration fetches data from the Sonic device every 2 minutes.
+Thanks to the webhook option, Sonic will provide live telemetry every second when water is flowing directly to Home Assistant.
```
[the remark this answers is not in the document]
Comment (adam-the-hero) [reply source/_integrations/watergate.markdown:76]: It is not required I guess.

**Review by frenck — APPROVED** [review]
Thanks, @adam-the-hero 👍 

../Frenck

**Pull request merged**